monitor/ConsoleReporter.js

```diff
@@ -72,16 +72,20 @@ define(function(require) {
 				// IE8 has console.log and JSON, so we can make a
 				// reasonably useful warn() from those.
 				// Credit to webpro (https://github.com/webpro) for this idea
-				if (typeof localConsole.log ==='function'
-					&& typeof JSON !== 'undefined') {
-					log = warn = function (x) {
-						if(typeof x !== 'string') {
+				// typeof localConsole.log will return 'object' in IE8, so can't test it with === 'function'
+				// Since this is more of a corner case for IE8, I'm ok to check it with !== 'undefined' to reduce complexity
+				if (typeof localConsole.log !== 'undefined' && typeof JSON !== 'undefined') {
```

Comment:
The code that uses only `typeof` would still fail because the execution internally goes in the following steps:
1. Access `localConsole.error`, [the line where it happens is in the collapsed code above, not visible in this PR by default](https://github.com/cujojs/when/pull/436/files#diff-2eb9f8e6caaab86409cabd1f4c79cac7R53). **While performing this operation IE8 can throw** sort of `Access denied.` exception AFAIR.
2. Apply the `typeof` operator to the result of the expression evaluated above. **This second step won't even get executed** if the previous one throws an exception.